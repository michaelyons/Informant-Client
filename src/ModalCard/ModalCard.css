h6 {
  color: black;
  background-color: azure;
}
