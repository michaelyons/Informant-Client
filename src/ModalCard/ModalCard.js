import React from 'react';

import PropTypes from 'prop-types';

import './ModalCard.css';

const ModalCard = ({ filingId, clientName, topic, register, date }) => {
  return (
    <div>
      <div className="card">
        <header className="card-header">
          <p className="card-header-title">Filing ID {filingId}</p>
          <button>
            Tweet This Issue
            <i className="fab fa-twitter" />
          </button>
          <p className="card-header-icon is-italic">Filed {date}</p>
        </header>
        <div className="media">
          <div className="card-content">
            <div className="media-content">
              <p className="title is-size-6">{topic}</p>
              <p className="subtitle">For Client {clientName}</p>
              <p className="subtitle">Registered by {register}</p>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

ModalCard.propTypes = {
  filingId: PropTypes.number,
  topic: PropTypes.string,
  clientName: PropTypes.string,
  register: PropTypes.string,
  date: PropTypes.string
};

export default ModalCard;
