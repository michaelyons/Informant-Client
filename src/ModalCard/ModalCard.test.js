import React from 'react';
import ModalCard from './ModalCard';
import { shallow } from 'enzyme';

describe('ModalCard Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ModalCard />);
  });

  describe('snapshot', () => {
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
