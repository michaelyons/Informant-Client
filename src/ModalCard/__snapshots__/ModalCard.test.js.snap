// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalCard Component snapshot should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card"
      >
        <header
          className="card-header"
        >
          <p
            className="card-header-title"
          >
            Filing ID 
          </p>
          <p
            className="card-header-icon is-italic"
          >
            Filed 
          </p>
        </header>
        <div
          className="media"
        >
          <div
            className="card-content"
          >
            <div
              className="media-content"
            >
              <p
                className="title is-size-6"
              />
              <p
                className="subtitle"
              >
                For Client 
              </p>
              <p
                className="subtitle"
              >
                Registered by 
              </p>
            </div>
          </div>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="card-header"
          >
            <p
              className="card-header-title"
            >
              Filing ID 
            </p>
            <p
              className="card-header-icon is-italic"
            >
              Filed 
            </p>
          </header>,
          <div
            className="media"
          >
            <div
              className="card-content"
            >
              <div
                className="media-content"
              >
                <p
                  className="title is-size-6"
                />
                <p
                  className="subtitle"
                >
                  For Client 
                </p>
                <p
                  className="subtitle"
                >
                  Registered by 
                </p>
              </div>
            </div>
          </div>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="card-header-title"
              >
                Filing ID 
              </p>,
              <p
                className="card-header-icon is-italic"
              >
                Filed 
              </p>,
            ],
            "className": "card-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Filing ID ",
                  undefined,
                ],
                "className": "card-header-title",
              },
              "ref": null,
              "rendered": Array [
                "Filing ID ",
                undefined,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Filed ",
                  undefined,
                ],
                "className": "card-header-icon is-italic",
              },
              "ref": null,
              "rendered": Array [
                "Filed ",
                undefined,
              ],
              "type": "p",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="card-content"
            >
              <div
                className="media-content"
              >
                <p
                  className="title is-size-6"
                />
                <p
                  className="subtitle"
                >
                  For Client 
                </p>
                <p
                  className="subtitle"
                >
                  Registered by 
                </p>
              </div>
            </div>,
            "className": "media",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="media-content"
              >
                <p
                  className="title is-size-6"
                />
                <p
                  className="subtitle"
                >
                  For Client 
                </p>
                <p
                  className="subtitle"
                >
                  Registered by 
                </p>
              </div>,
              "className": "card-content",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="title is-size-6"
                  />,
                  <p
                    className="subtitle"
                  >
                    For Client 
                  </p>,
                  <p
                    className="subtitle"
                  >
                    Registered by 
                  </p>,
                ],
                "className": "media-content",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "title is-size-6",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "For Client ",
                      undefined,
                    ],
                    "className": "subtitle",
                  },
                  "ref": null,
                  "rendered": Array [
                    "For Client ",
                    undefined,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Registered by ",
                      undefined,
                    ],
                    "className": "subtitle",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Registered by ",
                    undefined,
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card"
        >
          <header
            className="card-header"
          >
            <p
              className="card-header-title"
            >
              Filing ID 
            </p>
            <p
              className="card-header-icon is-italic"
            >
              Filed 
            </p>
          </header>
          <div
            className="media"
          >
            <div
              className="card-content"
            >
              <div
                className="media-content"
              >
                <p
                  className="title is-size-6"
                />
                <p
                  className="subtitle"
                >
                  For Client 
                </p>
                <p
                  className="subtitle"
                >
                  Registered by 
                </p>
              </div>
            </div>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <header
              className="card-header"
            >
              <p
                className="card-header-title"
              >
                Filing ID 
              </p>
              <p
                className="card-header-icon is-italic"
              >
                Filed 
              </p>
            </header>,
            <div
              className="media"
            >
              <div
                className="card-content"
              >
                <div
                  className="media-content"
                >
                  <p
                    className="title is-size-6"
                  />
                  <p
                    className="subtitle"
                  >
                    For Client 
                  </p>
                  <p
                    className="subtitle"
                  >
                    Registered by 
                  </p>
                </div>
              </div>
            </div>,
          ],
          "className": "card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="card-header-title"
                >
                  Filing ID 
                </p>,
                <p
                  className="card-header-icon is-italic"
                >
                  Filed 
                </p>,
              ],
              "className": "card-header",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Filing ID ",
                    undefined,
                  ],
                  "className": "card-header-title",
                },
                "ref": null,
                "rendered": Array [
                  "Filing ID ",
                  undefined,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Filed ",
                    undefined,
                  ],
                  "className": "card-header-icon is-italic",
                },
                "ref": null,
                "rendered": Array [
                  "Filed ",
                  undefined,
                ],
                "type": "p",
              },
            ],
            "type": "header",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="card-content"
              >
                <div
                  className="media-content"
                >
                  <p
                    className="title is-size-6"
                  />
                  <p
                    className="subtitle"
                  >
                    For Client 
                  </p>
                  <p
                    className="subtitle"
                  >
                    Registered by 
                  </p>
                </div>
              </div>,
              "className": "media",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="media-content"
                >
                  <p
                    className="title is-size-6"
                  />
                  <p
                    className="subtitle"
                  >
                    For Client 
                  </p>
                  <p
                    className="subtitle"
                  >
                    Registered by 
                  </p>
                </div>,
                "className": "card-content",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="title is-size-6"
                    />,
                    <p
                      className="subtitle"
                    >
                      For Client 
                    </p>,
                    <p
                      className="subtitle"
                    >
                      Registered by 
                    </p>,
                  ],
                  "className": "media-content",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "title is-size-6",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "For Client ",
                        undefined,
                      ],
                      "className": "subtitle",
                    },
                    "ref": null,
                    "rendered": Array [
                      "For Client ",
                      undefined,
                    ],
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Registered by ",
                        undefined,
                      ],
                      "className": "subtitle",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Registered by ",
                      undefined,
                    ],
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
